src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/HttpClientTransport.cs

```diff
@@ -133,7 +133,14 @@ protected internal override bool TryGetHeader(ReadOnlySpan<byte> name, out ReadO
             }
 
             // TODO (pri 1): is it ok to just call .Result here?
-            protected internal override Stream ResponseContentStream => _responseMessage.Content.ReadAsStreamAsync().Result;
+            protected internal override Stream ResponseContentStream {
+                get {
+                    if (_responseMessage.Content != null)
+                        return _responseMessage.Content.ReadAsStreamAsync().Result;
```

Comment:
Should we be using `.GetAwaiter().GetResult()` here instead of `.Result`?

Reply:
You may wish to consider simplifying this to something like:

```c#
protected internal override Stream ResponseContentStream =>
    _responseMessage?.Content?.ReadAsStreamAsync().GetAwaiter().GetResult();
```

Reply:
`GetAwaiter().GetResult()` is the way to go for blocking. But I think we should get rid of async blocking on this codepath.

Reply:
Well, that does not solve the main problem (of deadlocks). It only prevents any exceptions propagating from being wrapped in AggregateException. 
As you can see, there is a comment above this property to look into this. I am hoping that the method always returns synchronously, in which case Result would be ok.

Reply:
> I think we should get rid of async blocking on this codepath

Possibly. But it's kind of strange that just getting a stream is async. I wanted to talk to the HttpClient team to figure out why.

Reply:
Default implementation buffers content into a memory stream when calling `ReadAsStreamAsync` https://github.com/dotnet/corefx/blob/940ecee40b64ab93c92f87a7e8909771c013f1a5/src/System.Net.Http/src/System/Net/Http/HttpContent.cs#L450

Reply:
So probability of this call going async is very high.

Reply:
Well, then it seems like we will need to return our stream that lazily gets HttpClient stream inside its ReadAsync. But I think it needs to be a separate PR (per the existing comment)